### PR TITLE
Move `deer` to `libs`

### PR DESCRIPTION
**Pull request by kachkaev**
## 🌟 What is the purpose of this PR?

This PR moves `packages/libs/deer` to `libs/deer` as part of our migration to the new repo structure.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543021352032/1203324923411371/f) _(internal)_

## 🔍 What does this change?

- I created `libs/deer/package.json` to count `libs/deer` as a workspace. This enables license CI check for this folder and also allows us to configure tasks via Turborepo.

## 🐾 Next steps

- Move `packages/libs/error-stack` to `libs/error-stack` (when unblocked)

## 🛡 What tests cover this?

- CI

### CODE REVIEW

**Review by nonparibus — APPROVED** [review]
Legal LGTM. Requires code review still.

File: libs/deer/package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "deer",
+  "version": "0.0.0-private",
+  "private": true
+}
```
Comment (TimDiekmann) [inline libs/deer/package.json:5]: Could we [`exclude`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields) this file in `Cargo.toml` please?

**Review by TimDiekmann — CHANGES_REQUESTED** [review]

File: libs/deer/package.json (hunk 1 above)
Comment (kachkaev) [reply libs/deer/package.json:5]: Good point! Addressed in https://github.com/hashintel/hash/pull/1858/commits/e33821031577b682a1f5d36b6b69320360c02445

Comment (TimDiekmann) [reply libs/deer/package.json:5]: Thanks!

**Review by TimDiekmann — APPROVED** [review]

**Review by indietyp — APPROVED** [review]

**Review by indietyp — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]
Re-approval following changes

**Pull request merged**